src/compiler/config/validate-hydrated.ts

```diff
@@ -16,7 +16,7 @@ export const validateHydrated = (config: UnvalidatedConfig): HydratedFlag | unde
    *
    * See {@link HydratedFlag} for more details.
    */
-  if (config.hydratedFlag === null || config.hydratedFlag === false) {
```

Comment:
Feel free to push back on this.

Reply:
Same thoughts as above

Reply:
Similar to above, perhaps the check should become:
```diff
    *
    * See {@link HydratedFlag} for more details.
    */
-  if (config.hydratedFlag === null) {
+  if (config.hydratedFlag === null || (Array.isArray(config.hydratedFlag) && typeof config.hydratedFlag !== 'object')) {
     return undefined;
   }
```
Thoughts?

Reply:
Same here, but should all of the checks be `||`?